drivers/media/i2c/imx477.c

```diff
@@ -1316,7 +1352,8 @@ static int imx477_set_ctrl(struct v4l2_ctrl *ctrl)
 		break;
 	case V4L2_CID_EXPOSURE:
 		ret = imx477_write_reg(imx477, IMX477_REG_EXPOSURE,
-				       IMX477_REG_VALUE_16BIT, ctrl->val);
+				       IMX477_REG_VALUE_16BIT, ctrl->val /
+						(1 << imx477->long_exp_shift));
```

Comment:
```(ctrl->val >> imx477->long_exp_shift)```?

Reply:
I purposely did it with a divide, but in hindsight, the intention is not very clear :)
Will fix this an use shifts for all the other calculations.